lib/util/log.dart

```diff
@@ -119,3 +125,54 @@ class SentryBreadcrumbLogger {
         type: type));
   }
 }
+
+class BackgroundFileLogger {
+  static const _maxLines = 1000; // Maximum number of log lines
+
+  static final _logStreamController = StreamController<LogRecord>();
+  static late IOSink _fileSink;
+  static int _currentLineCount = 0;
+
+  static void initialize() async {
+    final logFile = await getLogFile();
+    _fileSink = logFile.openWrite(mode: FileMode.append);
+
+    _logStreamController.stream.listen(_writeLog);
+
+    // Start processing logs in the background
+    await for (var logRecord in _logStreamController.stream) {
```

Comment:
can it work without this, I saw there are no pending logs to add when initialize?

Reply:
but the `_logStreamController` will not be initialized again, so the data in the stream is still there and won't be dropped?

Reply:
Yes. That's why we need to flush the old items (if needed) here when running the `initialize()` again.

Reply:
btw, I remember `StreamController` prevents `listen` multiple times (unless it's a broadcast StreamController). So not sure it will throw exception when recall this function.